### ISSUE TITLE
[0.68.2] systemd user session destroys /mnt/wslg/runtime-dir

### ISSUE THREAD

**Issue by cerebrate** [microsoft/WSL#8918]
### Version

Microsoft Windows [Version 10.0.22000.978]

### WSL Version

- [X] WSL 2
- [ ] WSL 1

### Kernel Version

5.15.62.1-20220921-1-microsoft-custom-WSL2+

### Distro Version

Debian bookworm/sid

### Repro Steps

1. Start WSL with systemd enabled.
2. Observe that `/run/user/1000` has `/mnt/wslg/runtime-dir` mounted over it.
3. Start a systemd user session; for example, with systemd-machined installed, run `sudo machinectl shell avatar@host`

### Expected Behavior

The `/run/user/1000` and `/mnt/wslg/runtime-dir` directories should have identical contents, consisting of both user session-related and WSLg-related files and sockets.

### Actual Behavior

- The `/run/user/1000` directory now has only contents belonging to the user session, and none of those pertaining to WSLg.
- The `/mnt/wslg/runtime-dir` directory is now empty.

This appears to be the result of user-runtime-dir@.service not understanding that the user directory is already a mount point.

**Comment by cerebrate** [comment]
See also note to #8842.

**Comment by cerebrate** [comment]
As a workaround, this can be avoided by creating the following file as _/lib/systemd/system/user-runtime-dir@.service.d/override.conf_:

```
[Unit]
ConditionPathIsMountPoint=!/run/user/%i
```

Which will bypass attempting to create a new user runtime directory for any uids which already have one mounted.

**Comment by aki-k** [comment]
@cerebrate Amazing thank you very much. I had a hard time figuring this out (/mnt/wslg/runtime-dir being emptied)

**Comment by jacob-keller** [comment]
For anyone that ran into this issue, I had something similar happen where the wslg bind remount of /run/user/1000 was being done after my systemd-user service started. I fixed this by adding an override with

```
[Unit]
Requires=run-user-%i.mount
```

This forced the user service to start after the mount point, which allows systemctl --user to work, and appears to also work with wslg (both the regular files and the wayland socket appear, and all the contents are the same as the /mnt/wslg/run/user/1000.

Hope this helps anyone else, as I struggled with this for several days. Many other suggested fixes failed to work while preserving GUI application support.
